rust/kcl-lib/std/prelude.kcl

```diff
@@ -4,6 +4,225 @@
 
 export import * from "std::math"
 
+/// A number
+///
+/// May be signed or unsigned, an integer or decimal value.
+///
+/// You may see a number type with units, e.g., `number(mm)`. These are currently experimental.
+@(impl = primitive)
+export type number(unit)
```

Comment:
Was there a rationale for choosing parentheses for type parameters? I see that it's used for units of measure, but presumably they could be used for anything in the future. Is that right?

Reply:
The idea was to just use parens for units of measure so we could use <> for type parameters. But also it just kind of looks friendlier?

Reply:
`()` vs. `<>` vs. `[]` (brackets is used by Python, e.g. `list[str]`) is pretty arbitrary, in my opinion. I think I've seen ambiguity in some languages that have `<<` and `>>` infix operators, which is why some might prefer the others.

The thing that bothers me that I wish we could change is `bool[]`. It's a backwards holdover from C, in my opinion. Array is the type constructor, and bool is the parameter. `[bool]` makes so much more sense to me.

Reply:
we actually use both in various places and I have it on my todo list to change over to `[bool]`